### PR TITLE
ReaderZooming should remove the size of page footer when calculating zoom value

### PR DESCRIPTION
**Pull request by Hzj-jie**
This issue has been discussed in issue #2072.

### CODE REVIEW

File: frontend/apps/reader/modules/readerzooming.lua [hunk 1]
```diff
@@ -228,12 +228,19 @@ function ReaderZooming:getZoom(pageno)
         self.view:onBBoxUpdate(nil)
     end
     -- calculate zoom value:
-    local zoom_w = self.dimen.w / page_size.w
-    local zoom_h = self.dimen.h / page_size.h
-    if self.rotation % 180 ~= 0 then
+    local zoom_w = self.dimen.w
+    local zoom_h = self.dimen.h
+    if self.ui.view.footer_visible then
+        zoom_h = zoom_h - self.ui.view.footer.progress_text:getSize().h
```
Comment (houqp) [inline frontend/apps/reader/modules/readerzooming.lua]: Any reason not using `self.ui.view.footer:getSize()`?

Comment (chrox) [inline frontend/apps/reader/modules/readerzooming.lua]: I guess the top margin space is also good for rendering. I'm OK for both choices.

**Comment by Hzj-jie** [comment]
Sorry for the late response, yes, as Huang mentioned, we can still render in the top margin area of the footer.

**Comment by ilyats** [comment]
Doesn't seem to work now. Also, one should redo the scaling each time status bar visibility is flipped.

**Pull request merged**